code/datums/components/crafting/recipes.dm

```diff
@@ -26,6 +26,9 @@
 /datum/crafting_recipe/proc/check_requirements(mob/user, list/collected_requirements)
 	return TRUE
 
+/datum/crafting_recipe/proc/on_craft_completion(mob/user, atom/result)
```

Comment:
is that draft? I see this proc is called nowhere.

Reply:
line 896?????

Reply:
Line 896 doesn't call the proc. The way it is now will not do anything.